### DISCUSSION TITLE
Deployment approve/reject via python api supported?

### DISCUSSION THREAD

**Discussion by jbotello7381** [python-gitlab/python-gitlab#2252] [General]
I was playing with deployment and gitlab python library, and notice there's no method to approve or reject a deployment? i was able to do it via REST API from my python code, but not via the python wrapper.  I'm using 3.8.1 of the gitlab python library.

**Comment by JohnVillalovos** [comment]
Would be useful to have a link to the upstream GitLab API docs for what is missing. Thanks.

**Comment by jbotello7381** [comment]
Sorry, this is the endpoint
https://docs.gitlab.com/ee/ci/environments/deployment_approvals.html#approve-or-reject-a-deployment-using-the-api

**Comment by JohnVillalovos** [comment]
So it looks like it is this API:
https://docs.gitlab.com/ee/api/deployments.html#approve-or-reject-a-blocked-deployment

And it doesn't look like there is any support for that yet in:
https://github.com/python-gitlab/python-gitlab/blob/main/gitlab/v4/objects/deployments.py

Implementing it will look similar to how the "approve" method is done for a merge request.

https://github.com/python-gitlab/python-gitlab/blob/f67514e5ffdbe0141b91c88366ff5233e0293ca2/gitlab/v4/objects/merge_requests.py#L263-L290

**Reply by JohnVillalovos** [reply]
I opened https://github.com/python-gitlab/python-gitlab/issues/2253 as a RFE (Request For Enhancement)

**Reply by JohnVillalovos** [reply]
Created a PR: https://github.com/python-gitlab/python-gitlab/pull/2254

**Comment by jbotello7381** [comment]
I tested on my local environment with your PR and it works.   Thanks for the quick PR.

**Reply by JohnVillalovos** [reply]
> I tested on my local environment with your PR and it works. Thanks for the quick PR.

Thanks for the test! Glad to know it works.
